code/datums/lazy_template.dm

```diff
@@ -46,25 +44,55 @@
 	if(!load_path || !fexists(load_path))
 		CRASH("lazy template [type] has an invalid load_path: '[load_path]', check directory and map name!")
 
-	var/datum/map_template/loading = new(path = load_path, cache = TRUE)
-	if(!loading.cached_map)
+	var/datum/parsed_map/parsed_template = load_map(
+		file(load_path),
+		measure_only = TRUE,
+	)
+	if(isnull(parsed_template.parsed_bounds))
 		CRASH("Failed to cache lazy template for loading: '[key]'")
 
-	var/datum/turf_reservation/reservation = SSmapping.RequestBlockReservation(loading.width, loading.height)
+	var/width = parsed_template.parsed_bounds[MAP_MAXX] - parsed_template.parsed_bounds[MAP_MINX] + 1
+	var/height = parsed_template.parsed_bounds[MAP_MAXY] - parsed_template.parsed_bounds[MAP_MINY] + 1
+	var/datum/turf_reservation/reservation = SSmapping.request_turf_block_reservation(
+		width,
+		height,
+		parsed_template.parsed_bounds[MAP_MAXZ],
+	)
 	if(!reservation)
 		CRASH("Failed to reserve a block for lazy template: '[key]'")
 
-	if(!loading.load(coords2turf(reservation.bottom_left_coords)))
-		CRASH("Failed to load lazy template: '[key]'")
-	reservations += reservation
+	var/list/my_loaded_atoms = list()
+	for(var/z_idx in parsed_template.parsed_bounds[MAP_MAXZ] to 1 step -1)
+		var/turf/bottom_left = reservation.bottom_left_turfs[z_idx]
+		var/turf/top_right = reservation.top_right_turfs[z_idx]
+		load_map(
+			file(load_path),
+			bottom_left.x,
+			bottom_left.y,
+			bottom_left.z,
+			z_upper = z_idx,
+			z_lower = z_idx,
+		)
+		var/list/to_init = list()
+		for(var/turf/turf as anything in block(bottom_left, top_right))
+			to_init |= get_area(turf)
+			to_init |= turf
+			for(var/thing in turf.get_all_contents())
+				to_init |= thing
+
+		SSatoms.InitializeAtoms(to_init)
+		for(var/loaded_atom in to_init)
+			my_loaded_atoms |= loaded_atom
+
+	SEND_SIGNAL(src, COMSIG_LAZY_TEMPLATE_LOADED, my_loaded_atoms)
+	my_loaded_atoms.Cut() // We don't want the list to cause ref holding shenanigans due to ASYNC
```

Comment:
nothing between this line and the var going out of scope sleeps, so this should do nothing. what are you trying to guard against?

Reply:
people saving references to the list by using a signal handler that just does local_list = my_loaded_atoms

Reply:
isn't that a them problem? they could also copy it and cause the same problem. what's the point of trying to block against it like this